### PR TITLE
pci kubernetes µapp

### PR DESCRIPTION
**Pull request by frenautvh**
| Question         | Answer
| ---------------- | ---
| Branch?          | master
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | DTCORE-2441
| License          | BSD 3-Clause

## Description

add pci-kubernetes µapp

### CODE REVIEW

File: packages/manager/apps/pci-kubernetes/src/pages/list/useDatagridColumn.tsx [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+          </OsdsLink>
+        </DataGridTextCell>
+      ),
+      label: t('kube_list_name'),
+    },
+    {
+      id: 'id',
+      cell: (props: TKube) => (
+        <DataGridTextCell>
+          <OsdsClipboard
```
Comment (ppprevost) [inline packages/manager/apps/pci-kubernetes/src/pages/list/useDatagridColumn.tsx:31]: no tooltip "copied" is shown on click

File: packages/manager/apps/pci-kubernetes/src/components/service/ClusterAccessAndSecurity.component.tsx [hunk 2]
```diff
@@ -0,0 +1,315 @@
...
+            <OsdsText
+              className="mb-4"
+              size={ODS_TEXT_SIZE._200}
+              level={ODS_TEXT_LEVEL.heading}
+              color={ODS_THEME_COLOR_INTENT.text}
+              onClick={(event) => event.stopPropagation()}
+            >
+              {t('kube_service_file')}
+            </OsdsText>
+            <OsdsIcon
```
Comment (ppprevost) [inline packages/manager/apps/pci-kubernetes/src/components/service/ClusterAccessAndSecurity.component.tsx:175]: no pointer exclamation

File: packages/manager/apps/pci-kubernetes/src/pages/list/useDatagridColumn.tsx (hunk 1 above)
Comment (seven-amid) [reply packages/manager/apps/pci-kubernetes/src/pages/list/useDatagridColumn.tsx:31]: Replaced it by the one coming from manger-component

File: packages/manager/apps/pci-kubernetes/src/pages/list/List.page.tsx [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+            size={ODS_ICON_SIZE.xs}
+            name={ODS_ICON_NAME.PLUS}
+            className="mr-2"
+            color={ODS_THEME_COLOR_INTENT.primary}
+          />
+          {t('kube_list_cluster_create')}
+        </OsdsButton>
+
+        <div className="justify-between flex">
+          <OsdsSearchBar
```
[the remark this answers is not in the document]
Comment (seven-amid) [reply packages/manager/apps/pci-kubernetes/src/pages/list/List.page.tsx:107]: I confirm the issue is on Ods side

File: packages/manager/apps/pci-kubernetes/src/components/service/ClusterManagement.component.tsx [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+        />
+        <TileButton
+          title={t('kube_service_reset')}
+          isDisabled={isProcessing(kubeDetail?.status)}
+          href={hrefResetCluster}
+        />
+        <TileButton
+          title={tDetail('kube_common_create_node_pool')}
+          isDisabled={isProcessing(kubeDetail?.status)}
+          href=""
```
[the remark this answers is not in the document]
Comment (seven-amid) [reply packages/manager/apps/pci-kubernetes/src/components/service/ClusterManagement.component.tsx]: fixed

File: packages/manager/apps/pci-kubernetes/src/components/service/ClusterAccessAndSecurity.component.tsx (hunk 2 above)
Comment (seven-amid) [reply packages/manager/apps/pci-kubernetes/src/components/service/ClusterAccessAndSecurity.component.tsx:175]: Done

File: packages/manager/apps/pci-kubernetes/src/pages/new/steps/VersionStep.component.tsx
Comment (ppprevost) [inline packages/manager/apps/pci-kubernetes/src/pages/new/steps/VersionStep.component.tsx:1]: There is an issue on this file when i want to click on "edit this step": 

The state is empty -->  {!step.isLocked && <VersionSelector onSelectVersion={setVersion} />} the version selectory is not hydrated with the current version.

**Review by ppprevost — CHANGES_REQUESTED** [review]

File: packages/manager/apps/pci-rancher/src/components/layout-helpers/CreateRancher/CreateRancher.test.tsx [hunk 5]
```diff
@@ -156,15 +161,6 @@ describe('CreateRancher', () => {
       expect(button).toBeDisabled();
     });
 
-    it('Given that I am in a Discovery project, I should see the yellow banner inviting me to activate my project', async () => {
```
[the remark this answers is not in the document]
Comment (frenautvh) [reply packages/manager/apps/pci-rancher/src/components/layout-helpers/CreateRancher/CreateRancher.test.tsx]: we just added it back

File: packages/manager/modules/manager-pci-common/src/api/data/flavors.ts [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+export type TKubeFlavor = {
+  category: string;
+  gpus: number;
+  name: string;
+  ram: number;
+  state: 'available' | 'unavailable';
+  vCPUs: number;
+};
+
+export const getKubeFlavors = async (
```
[the remark this answers is not in the document]
Comment (frenautvh) [reply packages/manager/modules/manager-pci-common/src/api/data/flavors.ts:51]: it is used here : https://github.com/ovh/manager/blob/release%2Fpci-kubernetes/packages/manager/modules/pci/src/projects/project/kubernetes/service.js#L227

File: packages/manager/modules/manager-pci-common/src/api/data/availability.ts [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+    code: string;
+    regions: TRegion[];
+  }[];
+  products: {
+    name: string;
+    regions: TRegion[];
+  }[];
+};
+
+export const getProductAvailability = async (
```
[the remark this answers is not in the document]
Comment (frenautvh) [reply packages/manager/modules/manager-pci-common/src/api/data/availability.ts:22]: we just refactored it

File: packages/manager/modules/manager-pci-common/src/api/data/catalog.ts [hunk 8]
```diff
@@ -0,0 +1,130 @@
...
+  };
+  pricings: {
+    price: number;
+    type: string;
+    capacities: string[];
+  }[];
+};
+
+export const getCatalog = async (ovhSubsidiary: string): Promise<TCatalog> => {
+  const { data } = await v6.get<TCatalog>(
```
[the remark this answers is not in the document]
Comment (frenautvh) [reply packages/manager/modules/manager-pci-common/src/api/data/catalog.ts]: we just refactored it apart from database-analytics as more work will be required for this µapp

File: packages/manager/modules/manager-pci-common/src/api/data/instance.ts [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+  projectId: string,
+  instanceId: string,
+): Promise<TInstance> => {
+  const { data } = await v6.get<TInstance>(
+    `/cloud/project/${projectId}/instance/${instanceId}`,
+  );
+  return data;
+};
+
+export const getInstances = async (
```
[the remark this answers is not in the document]
Comment (frenautvh) [reply packages/manager/modules/manager-pci-common/src/api/data/instance.ts:38]: we just refactored it

File: packages/manager-react-components/src/components/action-banner/action-banner.spec.tsx [hunk 10]
```diff
@@ -38,8 +39,10 @@ describe('ActionBanner tests', () => {
       href,
     });
     const link = container.querySelector('osds-link');
+
+    console.log(link.getAttribute('href'));
```
Comment (anooparveti) [inline packages/manager-react-components/src/components/action-banner/action-banner.spec.tsx:43]: Remove this console statement.

**Review by anooparveti — CHANGES_REQUESTED** [review]
I dont see the documentation updated to handle the discovery project on Manager pages.

File: packages/manager-react-components/src/components/action-banner/action-banner.spec.tsx (hunk 10 above)
Comment (frenautvh) [reply packages/manager-react-components/src/components/action-banner/action-banner.spec.tsx]: done thx

**Pull request merged**